### DISCUSSION TITLE
Flushing the Text-to-Speech Server Buffer Mid-Stream

### DISCUSSION THREAD

**Discussion by Stillerman** [fishaudio/fish-speech#742] [Q&A] [answered]
We’re using the [example code](https://docs.fish.audio/text-to-speech/text-to-speech-ws#example-usage-with-openai-mpv) from the documentation, which allows us to stream text asynchronously and play the audio as it’s ready. This functionality is fantastic! However, in our use case, we occasionally encounter short pieces of text that we would like to process and play immediately.

From the documentation:

_"There is a text buffer on the server side. Only when this buffer reaches a certain size will an audio event be generated. Sending a stop event will force the buffer to be flushed, return an audio event, and end the session."_

In our scenario, we have cases like:
"Hi" **<10-second pause>** "How's it going?"

Is there a way to trigger the buffer to flush within that 10-second pause before the buffer reaches its full size? While we could send a stop event to force the buffer to flush and then restart, this disrupts the input stream and introduces extra complexity.

It would be incredibly helpful if there were a way to send a command like flush to process the current buffer without ending the session. Is there an existing way to achieve this, or would this require additional support in the API?

Thanks in advance for your help!

**Comment by Stillerman** [comment]
@leng-yue any thoughts on this?

**Reply by leng-yue** [reply, marked as answer]
Thanks for the feedback, we have added a `flush` event :)
